skimage/measure/_regionprops_utils.py

```diff
@@ -326,3 +327,38 @@ def perimeter_crofton(image, directions=4):
 
     total_perimeter = coefs @ h
     return total_perimeter
+
+
+def _normalize_spacing(spacing, ndims):
+    """Normalize spacing parameter.
+
+    The `spacing` parameter should be a sequence of numbers
+    that is the same length as the number of dimensions
+    in the image.
+    If `spacing` is a scalar (single number),
+    then assume spacing is the same in all dimensions.
+
+    Parameters
+    ---------
+    spacing : Any
+        User-provided `spacing` keyword.
+    ndims: int
+        Number of dimensions in the image.
+
+    Returns
+    -------
+    spacing : array
+        Corrected spacing, if possible.
+    """
+    spacing = np.array(spacing)
+    if spacing.shape == ():
+        spacing = np.broadcast_to(spacing, shape=(ndims,))
+    elif spacing.shape != (ndims,):
+        raise ValueError(
+            f"spacing isn't a scalar nor a sequence of shape {(ndims,)}, got {spacing}"
+        )
+    if not all(isinstance(s, Real) for s in spacing):
```

Comment:
@tonyreina, I took the liberty to change this from `elif` to `if` and also patched / refactored a few of the tests. If you don't agree with something in https://github.com/scikit-image/scikit-image/pull/6907/commits/dcd7d73d49d622c582331ac326b64d37f4e2246c please feel free to revert or comment! :pray:

Reply:
I just merged our `main` branch with this one. I'm hoping that will fix the weird failure on CircleCI. :crossed_fingers:

Reply:
Looks like it did the trick